### PR TITLE
SimpleGADriver - Mutation rate and crossover rate as driver options

### PR DESCRIPTION
**Pull request by onodip**
Mutation rate and crossover rate added to the options dictionary, so it is more accessible to the user.  Mutation rate will be calculated as previously if it is not set by the user. Crossover defaults to the same value (0.5), as previously.

Minor other changes:
GeneticAlgorithm() changed to GeneticAlgorithm(object) ("new type" class), to be consistent with other OpenMDAO classes.

### CODE REVIEW

**Comment by onodip** [comment]
Any better ideas, how to deal with options, which do not have a default value, but are calculated, if not specified by the user? Now I used None as default, and a conditional in the code. Is this the correct way?

**Comment by Kenneth-T-Moore** [comment]
That is the way that I've done it in the past. The options dictionary has an "allow_none" attribute too, so that you can still enforce type and bounds while also allowing None to be specified.

File: openmdao/drivers/genetic_algorithm_driver.py [hunk 1]
```diff
@@ -437,8 +448,9 @@ def execute_ga(self, vlb, vub, bits, pop_size, max_gen, random_state):
...
         fitness = np.zeros((self.npop, ))
 
-        Pc = 0.5
-        Pm = (self.lchrom + 1.0) / (2.0 * pop_size * np.sum(bits))
+        # If mutation rate is not provided as input
+        if Pm is None:
+            Pm = (self.lchrom + 1.0) / (2.0 * pop_size * np.sum(bits))
         elite = self.elite
 
         # TODO: from an user-supplied intial population
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/genetic_algorithm_driver.py]: I would probably just add a small test where you set these (just to verify that options in the api can be set.) Probably doesn't need to be a big test.

Comment (onodip) [reply openmdao/drivers/genetic_algorithm_driver.py]: Test added.

File: openmdao/docs/Makefile [hunk 2]
```diff
@@ -13,10 +13,10 @@ ifeq ($(shell which $(SPHINXBUILD) >/dev/null 2>&1; echo $$?), 1)
 $(error The '$(SPHINXBUILD)' command was not found. Make sure you have Sphinx installed, then set the SPHINXBUILD environment variable to point to the full path of the '$(SPHINXBUILD)' executable. Alternatively you can add the directory with the executable to your PATH. If you don't have Sphinx installed, grab it from http://sphinx-doc.org/)
 endif
 
-SPHINX_VERSION := $(shell $(SPHINXBUILD) --version | grep -o -E "[0-9\.]+")
-ifeq ($(shell expr $(SPHINX_VERSION) \< 1.5), 1)
-$(error Version 1.5 or later of Sphinx is required, but $(SPHINX_VERSION) was found.  Upgrade via 'pip install --upgrade sphinx' or get the latest from http://sphinx-doc.org/)
-endif
+# SPHINX_VERSION := $(shell $(SPHINXBUILD) --version | grep -o -E "[0-9\.]+")
```
[the remark this answers is not in the document]
Comment (onodip) [reply openmdao/docs/Makefile]: Yes, completely unintentional, it was changed back in the last commit.

[the remark this answers is not in the document]
Comment (kmarsteller) [reply openmdao/docs/Makefile]: I am still seeing the Sphinx version check at the top as commented out in latest.  Saw you brought back the other stuff later in the file?  Maybe I am not looking at latest...

**Pull request merged**